File: Java/dependency-injection/service-locator/src/main/java/ServiceLocatorAntipattern/ApplicationRoot.java
package ServiceLocatorAntipattern;

import ServiceLocatorAntipattern.Core.AcmeProcessingWorkflow;
import ServiceLocatorAntipattern.Core.ProcessingWorkflow;
import ServiceLocatorAntipattern.Core.TeleComSystem;
import ServiceLocatorAntipattern.InMessages.NullMessage;
import ServiceLocatorAntipattern.InMessages.StartMessage;
import ServiceLocatorAntipattern.InMessages.StopMessage;
import ServiceLocatorAntipattern.Inbound.BinaryParsing;
import ServiceLocatorAntipattern.Inbound.BinaryUdpInbound;
import ServiceLocatorAntipattern.Inbound.IInbound;
import ServiceLocatorAntipattern.Inbound.PacketParsing;
import ServiceLocatorAntipattern.Outbound.*;
import ServiceLocatorAntipattern.Services.*;
import org.picocontainer.ComponentMonitor;
import org.picocontainer.DefaultPicoContainer;
import org.picocontainer.MutablePicoContainer;
import org.picocontainer.lifecycle.ReflectionLifecycleStrategy;
import org.picocontainer.monitors.NullComponentMonitor;

import java.util.Random;

import static org.picocontainer.Characteristics.CACHE;
import static org.picocontainer.Characteristics.NO_CACHE;


public class ApplicationRoot {
  public static final MutablePicoContainer context = createContainer();

  private static MutablePicoContainer createContainer() {
    ComponentMonitor monitor = new NullComponentMonitor();
    MutablePicoContainer context = new DefaultPicoContainer(
        monitor,
        new ReflectionLifecycleStrategy(monitor),
        null);
    context.as(CACHE)
        .addComponent(IRepository.class, MsSqlBasedRepository.class)
        .addComponent(IAuthorization.class, ActiveDirectoryBasedAuthorization.class)
        .addComponent(ProcessingWorkflow.class, AcmeProcessingWorkflow.class)
        .addComponent(OutputSocket.class, TcpSocket.class)
        .addComponent(Outbound.class, MessageOutbound.class)
        .addComponent(PacketParsing.class, BinaryParsing.class)
        .addComponent(IInbound.class, BinaryUdpInbound.class)
        // forgot about this... context.addComponent(IMarshalling.class, XmlMarshalling.class);
        .addComponent(SqlDataDestination.class);
    context.as(NO_CACHE)
        .addComponent(IOutboundMessage.class, OutboundMessage.class)
        .addComponent(Random.class)
        .addComponent(NullMessage.class)
        .addComponent(StartMessage.class)
        .addComponent(StopMessage.class);

    return context;
  }

  public static void Main(String[] args) {
    try {
      TeleComSystem sys = new TeleComSystem(); //uses container inside, but should be resolved itself!
      sys.start();

    } finally {
      context.dispose();
    }
  }
}

File: CSharp/DependencyInjection/DependencyInjection/ConformingContainerAntipattern/_3_ConformingContainer/ApplicationRoot.cs
using System;
using ConformingContainerAntipattern._3_ConformingContainer.Core;
using ConformingContainerAntipattern._3_ConformingContainer.InMessages;
using ConformingContainerAntipattern._3_ConformingContainer.Inbound;
using ConformingContainerAntipattern._3_ConformingContainer.Outbound;
using ConformingContainerAntipattern._3_ConformingContainer.Services;

namespace ConformingContainerAntipattern._3_ConformingContainer
{
    public static class ApplicationRoot
    {
      static ApplicationRoot()
      {
        Context.For<IProcessingWorkflow>().UseAlwaysTheSame<AcmeProcessingWorkflow>();
        Context.For<IInbound>().UseAlwaysTheSame<BinaryUdpInbound>();
        Context.For<IInputSocket>().UseAlwaysTheSame<UdpSocket>();
        Context.For<IOutputSocket>().UseAlwaysTheSame<TcpSocket>();
        Context.For<IPacketParsing>().UseAlwaysTheSame<BinaryParsing>();
        Context.For<IOutbound>().UseAlwaysTheSame<Outbound.Outbound>();
        Context.For<IAuthorization>().UseAlwaysTheSame<ActiveDirectoryBasedAuthorization>();
        Context.For<IRepository>().UseAlwaysTheSame<MsSqlBasedRepository>();
        Context.For<IMarshalling>().UseAlwaysTheSame<XmlMarshalling>();

        Context.UseAlwaysTheSame<SqlDataDestination>();

        //not container controlled
        Context.For<IOutboundMessage>().UseEachTimeNew<OutboundMessage>();
        Context.For<Random>().UseInstancesCreatedWith(container => new Random());
        Context.UseEachTimeNew<NullMessage>();
        Context.UseEachTimeNew<StartMessage>();
        Context.UseEachTimeNew<StopMessage>();
      }

      public static void Main2(string[] args)
      {
        try
        {
          var sys = new TeleComSystem(); //uses container inside, but should be resolved itself!
          sys.Start();

        }
        finally
        {
          Context.Dispose();
        }
      }

      //simplified
      public static readonly ConformingContainer Context = new ConformingContainer();
    }


}
